Examples/MAX32665/BLE_otas/project.mk

```diff
@@ -21,5 +21,16 @@ INIT_OBSERVER = 0
 # Optimize for size
 MXC_OPTIMIZE_CFLAGS = -Os
 
-# Use local linkerfile
-LINKERFILE = ota.ld
+AUTOSEARCH=0
+VPATH += .
+SRCS += $(wildcard $(addsuffix /*.c, $(VPATH)))
```

Comment:
@EdwinFairchild @kevin-gillespie for my own curiosity why is this necessary?  Autosearch should hit the root directory of the project for all source code by default (see [default behavior](https://analog-devices-msdk.github.io/msdk/USERGUIDE/#default-build-behavior))

Reply:
further down I exclude a file, and since project.mk is parsed before the rest of the make file, atuosearch will go in and add the file we want to exclude again. 
I did change this to look like this though

```

USE_INTERNAL_FLASH ?=0
ifeq ($(USE_INTERNAL_FLASH), 1)
LINKERFILE = ota_internal_mem.ld
SRCS += wdxs_file_int.c
else
LINKERFILE = ota_external_mem.ld
SRCS += wdxs_file_ext.c
endif

```

Reply:
Ah ok.  You could also consider a compiler definition:

```Makefile
# project.mk

# (Leave autosearch enabled, then...)
USE_INTERNAL_FLASH ?=0
ifeq ($(USE_INTERNAL_FLASH), 1)
LINKERFILE = ota_internal_mem.ld
PROJ_CFLAGS+=-DUSE_INTERNAL_FLASH
# ^ Add compiler definition with -D[variable name]
else
LINKERFILE = ota_external_mem.ld
endif
```

```C
// wdxs_file.c

#ifdef USE_INTERNAL_FLASH
// (New internal flash stuff...)
#else
// (Old external flash stuff...)
#endif
```

This option may be better for reducing code duplication and Makefile complexity.

Reply:
@Jake-Carter thats what I had originally but the file jwas so choppy it was hard to look at honestly.